test/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -0,0 +1,158 @@
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h"
+#include "envoy/extensions/transport_sockets/tls/v3/cert.pb.h"
+
+#include "extensions/transport_sockets/tls/context_config_impl.h"
+#include "extensions/transport_sockets/tls/ssl_socket.h"
+
+#include "test/integration/http_integration.h"
+#include "test/integration/ssl_utility.h"
+
+namespace Envoy {
+namespace {
+
+class ProxyFilterIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                   public Event::TestUsingSimulatedTime,
+                                   public HttpIntegrationTest {
+public:
+  ProxyFilterIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam(),
+                            ConfigHelper::TCP_PROXY_CONFIG) {}
+
+  static std::string ipVersionToDnsFamily(Network::Address::IpVersion version) {
+    switch (version) {
+    case Network::Address::IpVersion::v4:
+      return "V4_ONLY";
+    case Network::Address::IpVersion::v6:
+      return "V6_ONLY";
+    }
+
+    // This seems to be needed on the coverage build for some reason.
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+
+  void setup(uint64_t max_hosts = 1024) {
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP1);
+
+    config_helper_.addListenerFilter("name: envoy.filters.listener.tls_inspector");
```

Comment:
should there be a test that shows that this doesn't work without the tls inspector listener filter?

Reply:
I don't think so, there's too many cases that a missing configuration doesn't work (i.e. the filter without cluster etc.)